packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -493,15 +466,29 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		localId: string,
 		blob: ArrayBufferLike,
 	): Promise<ICreateBlobResponse | void> {
-		return PerformanceEvent.timedExecAsync(
-			this.mc.logger,
-			{ eventName: "createBlob" },
+		return runWithRetry(
 			async () => this.getStorage().createBlob(blob),
-			{ end: true, cancel: this.runtime.connected ? "error" : "generic" },
-		).then(
-			(response) => this.onUploadResolve(localId, response),
-			async (err) => this.onUploadReject(localId, err),
-		);
+			"createBlob",
+			this.mc.logger,
+			{
+				cancel: this.pendingBlobs.get(localId)?.abortSignal,
+			},
+		)
+			.then((response) => this.onUploadResolve(localId, response))
+			.catch((err) => {
```

Comment:
i think this code should go on the inside of the retry loop, not outside it. something like 

```
async () => {
		try{
   				 return await this.getStorage().createBlob(blob);
		}catch(error){
				const entry = this.pendingBlobs.get(localId);
				assert(
					!!entry,
					0x387 /* Must have pending blob entry for blob which failed to upload */,
				);
				if (entry.opsent && !canRetryOnError(error)) {
					/// this is the important part
					throw isFluidError(e) ? e.addTelemetryProperties({canRetry: true}) : wrapError(e, ()=>createErrorObjectWithProps("createBlob");
				}
				entry.handleP.reject(err);
				this.deletePendingBlob(localId);
				throw error;
				
		}
}
```

Reply:
the problem with the existing code is there is no retry delay if the error is not retriable, so this change makes all errors retriable when we have to succeed.